src/npcmove.cpp

```diff
@@ -2818,3 +2865,30 @@ bool npc::complain()
     // TODO: Complain about hunger and thirst, when NPCs can have those
     return false;
 }
+
+void npc::do_reload( item &it )
+{
+    auto usable_ammo = find_usable_ammo( it );
+
+    if( !usable_ammo ) {
+        debugmsg( "do_reload failed: no usable ammo" );
+        return;
+    }
+
+    int reload_time = item_reload_cost( it, *usable_ammo );
```

Comment:
you need to pass `qty` here to get the correct (reduced) times for partially loaded items